api/client/swarm/opts.go

```diff
@@ -99,27 +101,31 @@ func (o *AutoAcceptOption) String() string {
 }
 
 // Set sets a new value on this option
-func (o *AutoAcceptOption) Set(value string) error {
-	value = strings.ToUpper(value)
-	switch value {
-	case "", "NONE":
-		if accept, ok := o.values[WORKER]; ok && accept {
-			return fmt.Errorf("value NONE is incompatible with %s", WORKER)
-		}
-		if accept, ok := o.values[MANAGER]; ok && accept {
-			return fmt.Errorf("value NONE is incompatible with %s", MANAGER)
-		}
-		o.values[WORKER] = false
-		o.values[MANAGER] = false
-	case WORKER, MANAGER:
-		if accept, ok := o.values[value]; ok && !accept {
-			return fmt.Errorf("value NONE is incompatible with %s", value)
+func (o *AutoAcceptOption) Set(acceptValues string) error {
+	for _, value := range strings.Split(acceptValues, ",") {
+		value = strings.ToUpper(value)
+		switch value {
+		case "", NONE:
+			// NONE must stand alone, so if any non-NONE setting exists, conflict
+			if !o.values[NONE] && len(o.values) > 0 {
```

Comment:
I think it makes most sense for it to be below the loop because otherwise you have to check for conflicts both in the `none` case (in case `none` comes after a different value) and in the other cases (in case they come after `none`). If the check is below the loop, only a single check is needed.